code/__DEFINES/~~bubber_defines/misc.dm

```diff
@@ -0,0 +1,2 @@
+//Automatic punctuation
+#define ENDING_PUNCT list(".", "-","?", "!","~", "+", "|", "_")
```

Comment:
I feel like this can be done better with regex, specifically with the phrase

``^[^\*].[^.!?~\+\|\_]+$``

As this will prevent it being used on emotes.

Initialize the global datum somewhere early.

``GLOBAL_DATUM_INIT(has_eol_punctuation, /regex, regex(@^[^\*].[^.!?~\+\|\_]+$))``

Then use ``findtext(chat_message, GLOB.has_eol_punctuation)`` instead of copytext.

Should work. Further testing may be needed.